DataFormats/OnlineMetaData/test/onlineMetaDataRecord_t.cpp

```diff
@@ -0,0 +1,141 @@
+#include <cppunit/extensions/HelperMacros.h>
+#include "DataFormats/OnlineMetaData/interface/DCSRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineBeamSpotRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineLuminosityRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineMetaDataRaw.h"
+
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <string.h>
+#include <vector>
+
+
+class TestOnlineMetaDataRecord: public CppUnit::TestFixture
+{
+  CPPUNIT_TEST_SUITE(TestOnlineMetaDataRecord);
+  CPPUNIT_TEST(testDCSRecord);
+  CPPUNIT_TEST(testOnlineBeamSpotRecord);
+  CPPUNIT_TEST(testOnlineLuminosityRecord);
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+
+  void setUp() override;
+  void tearDown() override {}
+
+  void testDCSRecord();
+  void testOnlineBeamSpotRecord();
+  void testOnlineLuminosityRecord();
+
+private:
+  const std::string dumpFileName = "dump_run000001_event00020137_fed1022.txt";
+  std::vector<uint32_t> data;
+  onlineMetaData::Data_v1 const* onlineMetaData;
+  //OnlineMetaDataRecord onlineMetaDataRecord;
+};
+
+///registration of the test so that the runner can find it
+CPPUNIT_TEST_SUITE_REGISTRATION(TestOnlineMetaDataRecord);
+
+
+void TestOnlineMetaDataRecord::setUp() {
+
+  if ( data.empty() ) {
+    char* cmsswBase;
+    cmsswBase = getenv("CMSSW_BASE");
+    std::ostringstream dumpFileStr;
+    dumpFileStr << cmsswBase << "/src/DataFormats/OnlineMetaData/test/" << dumpFileName;
```

Comment:
this will fail during IB tests and project dev area (where this package is not checkout out). You need to use something like this https://github.com/cms-sw/cmssw/blob/bab6aa259dcd4c5be70c84fd227203943f7befe6/FWCore/Catalog/test/FileLocator_t.cpp#L52

i.e. if exists CMSSW_BASE+yourfile then use it otherwise use CMSSW_RELEASE_BASE+yourfile

Reply:
Thanks, will be fixed.